### ISSUE TITLE
test_fine_circular_buffer fails with gcc 12.1+

### ISSUE THREAD

**Issue by abulimov** [Orolia2s/disciplining-minipod#28]
when packaging latest release for Fedora/EPEL I've discovered that *test_fine_circular_buffer* fails with newer gcc. Test passes when built with gcc 8.5 that is in EPEL.

```
test-fine-circular-buffer: /builddir/build/BUILD/disciplining-minipod-3.2.0/test/test_fine_circular_buffer.c:83: main: Assertion `compare_float(get_fine_from_table(&state, 50.000001), state.estimated_equilibrium_ES)' failed. 
```

The same test also fails on ppc64le architecture even on Centos8 - https://koji.fedoraproject.org/koji/getfile?taskID=87838272&volume=DEFAULT&name=build.log&offset=-4000

**Comment by CharlesParent** [comment]
Hi @abulimov,
Could you send us the full log of both compilation and test please ?

**Comment by abulimov** [comment]
@CharlesParent not sure how helpful this is, but sure - here is the build + run log for this test - https://gist.github.com/abulimov/36c503436bb59b369a5f580b8843f354, collected on bog standard Fedora 36.

**Comment by CharlesParent** [comment]
Hi @abulimov, Issue should be fixed in version 3.2.2
Please let me know if that is not the case

**Comment by abulimov** [comment]
Thank you @CharlesParent, I've confirmed that the issue is now fixed.
